### DISCUSSION TITLE
Why do GitHub-published actions use (and move!) tags for versions?

### DISCUSSION THREAD

**Discussion by nedbat** [community/community#35011] [Actions]
Looking at the actions that GitHub published at https://github.com/actions, they do something unexpected.  Look at the tags in https://github.com/actions/setup-node/tags for example.  Each specific version (like "v3.5.0") is a tag. That's a typical use of tags: mark a definite point in the commit history, and never change it.  But "v3" refers to the same commit.  It gets updated when a new v3.x version is released.

Why use tags this way?  A more usual git practice would be for v3.5.0 to be a tag, and for v3 to be a branch. Even better, use the name "v3.x" for the branch to be very explicit that it will change over time.

With the increasing focus on security, wouldn't it be better to stick with git practice and let tags be immutable?

**Comment by sohmc** [comment]
The moving of major and minor tags is compatible with SemVer.  v3 is meant to mean the latest version of v3.x.x and v3.5 means the latest version of v3.5.x. 

Any person who wishes not to use the latest can pin down to the commit SHA if they wish so not sure if I'd call this a security issue.  In fact, I'd argue the opposite: if I know there is a bug in 3.4.2 and you are pinned to that version, I can crawl GitHub and find your GitHub action call and exploit it.  But if you were just pinned to 3.4 or even 3 you'd be fine.

git allows you to move git tags and I don't think GitHub is doing anything special here.
